# Cleansing Data

#### Import Library

In [2]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [3]:
# Memuat value dari file .env
load_dotenv()

mongo_uri = os.getenv('URL_SANDY')

#### Import dataset dari MongoDB

In [2]:
# Koneksi ke MongoDB
client = MongoClient(mongo_uri)

# Pilih database dan koleksi
db = client['youtube_rewind_indonesia']
collection = db['collect_2015']

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [3]:
df

_id        video                      author  \
0      664b4774d0e4a8c2ded86821  fNX6Gu2F7A8                   @wibu4353   
1      664b4774d0e4a8c2ded86822  fNX6Gu2F7A8  @ginawardahtulhumairoh3722   
2      664b4774d0e4a8c2ded86823  fNX6Gu2F7A8                @alfiqri5913   
3      664b4774d0e4a8c2ded86824  fNX6Gu2F7A8          @dastananindra1566   
4      664b4774d0e4a8c2ded86825  fNX6Gu2F7A8                   @farisnst   
...                         ...          ...                         ...   
28403  664b4774d0e4a8c2ded8d714  fNX6Gu2F7A8           @GantplayBlogspot   
28404  664b4774d0e4a8c2ded8d715  fNX6Gu2F7A8           @ramandahafiz6293   
28405  664b4774d0e4a8c2ded8d716  fNX6Gu2F7A8                    @e4rtz84   
28406  664b4774d0e4a8c2ded8d717  fNX6Gu2F7A8      @pricillaoktaviani2376   
28407  664b4774d0e4a8c2ded8d718  fNX6Gu2F7A8           @abayartsgame4609   

                 updated_at  like_count  \
0      2024-04-15T09:56:11Z           0   
1      2024-04-14T14:43:44Z           0   
2      2024-04-11T13:56:04Z           1   
3      2024-04-07T06:20:23Z           1   
4      2024-04-01T22:02:47Z           1   
...                     ...         ...   
28403  2015-12-11T06:02:11Z           0   
28404  2015-12-11T06:02:07Z           0   
28405  2015-12-11T06:02:05Z           1   
28406  2015-12-11T06:02:03Z           0   
28407  2015-12-11T06:02:01Z           1   

                                                text  public  
0                                    salam dari 2024    True  
1             Hahaha...btw niii udah 2024,GK kerasa😢    True  
2                                           the best    True  
3                                Sorry telat 9 tahun    True  
4      Ah gila udah mau 9 tahun masih apal lagu nya😂    True  
...                                              ...     ...  
28403                                      Mantep :v    True  
28404                                         Boom !    True  
28405                        anjing canggih waaahhhh    True  
28406                                          firat    True  
28407                                              1    True  

[28408 rows x 7 columns]

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['text'])

# Melihat banyak data yang duplikat
df.duplicated().sum()
df.shape

(26480, 7)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['author'])

# Melihat banyak data yang duplikat
df.duplicated().sum()
df.shape

(24311, 7)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan text

In [8]:
df = df[['_id', 'text']]
df

_id                                           text
0      664b4774d0e4a8c2ded86821                                salam dari 2024
1      664b4774d0e4a8c2ded86822         Hahaha...btw niii udah 2024,GK kerasa😢
2      664b4774d0e4a8c2ded86823                                       the best
3      664b4774d0e4a8c2ded86824                            Sorry telat 9 tahun
4      664b4774d0e4a8c2ded86825  Ah gila udah mau 9 tahun masih apal lagu nya😂
...                         ...                                            ...
28395  664b4774d0e4a8c2ded8d70c                                          woeee
28396  664b4774d0e4a8c2ded8d70d                                              1
28402  664b4774d0e4a8c2ded8d713                                    =))) kerenn
28403  664b4774d0e4a8c2ded8d714                                      Mantep :v
28405  664b4774d0e4a8c2ded8d716                        anjing canggih waaahhhh

[24311 rows x 2 columns]

#### Menghapus nilai yang kosong

In [10]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id     0
text    0
dtype: int64

Dari yang awalnya berjumlah 28408 row berkurang menjadi 24311 row

#### Mengimpan Clean Comments ke MongoDB Atlas

In [14]:
# Koneksi ke MongoDB
client = MongoClient("mongodb+srv://ekaputrajuniawan:9yyb7lj6k557n45i@cluster0.dboezoc.mongodb.net/")
db = client["youtube_rewind_indonesia"]
collection = db["cleansing_2015"]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Atlas.")

Data berhasil diimpor ke MongoDB Atlas.
